src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -87,20 +118,60 @@ private static bool ApplyChangesFromTC25ThroughTC28(JObject sarifLog)
                         modifiedLog = true;
                     }
 
+                    // Logical locations are now an array. We will first transform the previous
+                    // dictionary into the arrays form. We will retain the index for each
+                    // transformed logical location. Later, we will associate these indices
+                    // with results, if applicable.
+                    var logicalLocations = run["logicalLocations"] as JObject;
+                    Dictionary<LogicalLocation, int> logicalLocationToIndexMap = null;
+
+                    if (logicalLocations != null)
+                    {
+                        logicalLocationToIndexMap = new Dictionary<LogicalLocation, int>(LogicalLocation.ValueComparer);
+                        var jObjectToIndexMap = new Dictionary<JObject, int>();
+
+                        run["logicalLocations"] =
+                            ConstructLogicalLocationsArray(
+                                logicalLocations,
+                                jObjectToIndexMap,
```

Comment:
>jObjectToIndexMap [](start = 32, length = 17)

NIT: We never use this local variable after passing the empty dictionary into `ConstructLogicalLocationsArray`, so we may as well let that method create the dictionary.

Reply:
Ok, I see that this method can be called recursively, so it has to start off with the dictionary object.

---
In reply to: [243855979](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243855979) [](ancestors = 243855979)